src/BSON.jl

```diff
@@ -5,7 +5,20 @@ export bson
 
 using Core: SimpleVector, TypeName
 
-const BSONDict = Dict{Symbol,Any}
+struct BSONDict <: AbstractDict{Symbol,Any}
+  d::Dict{Symbol,Any}
+  BSONDict(v...) = new(Dict{Symbol,Any}(v...))
+end
+# implement some of the needed Dict methods
+Base.length(bd::BSONDict) = length(bd.d)
+Base.isempty(bd::BSONDict) = length(bd.d)==0
```

Comment:
Should `isempty` be a direct delegation?

Reply:
yes